Shared/Samples/Show viewshed from point on map/ShowViewshedFromPointOnMapView.swift

```diff
@@ -0,0 +1,183 @@
+        return inputGraphicsOverlay
+    }()
+    /// The graphics overlay that displays the viewshed for the tapped location.
+    @State private var resultGraphicsOverlay = {
+        let resultGraphicsOverlay = GraphicsOverlay()
+        let fillColor = UIColor(
+            red: 226 / 255.0,
+            green: 119 / 255.0,
+            blue: 40 / 255,
+            alpha: 120 / 255.0
+        )
+        let fillSymbol = SimpleFillSymbol(
+            style: .solid,
+            color: fillColor,
+            outline: nil
+        )
+        let resultRenderer = SimpleRenderer(symbol: fillSymbol)
+        resultGraphicsOverlay.renderer = resultRenderer
+        return resultGraphicsOverlay
+    }()
+    /// The current draw status of the map.
+    @State private var currentDrawStatus: Bool = false
+    /// The processing task that references the online resource through the url.
+    @State private var geoprocessingTask: GeoprocessingTask = {
+        let geoprocess = GeoprocessingTask(url: .viewshedURL)
+        return geoprocess
+    }()
+    /// Handles the execution of the geoprocessing task and gets the result.
+    @State private var geoprocessingJob: GeoprocessingJob?
+    /// Sets map's initial viewpoint to Vanoise National Park in France.
+    @State private var map: Map = {
+        let map = Map(basemapStyle: .arcGISTopographicBase)
+        map.initialViewpoint = Viewpoint(
+            center: Point(latitude: 45.3790902612337, longitude: 6.84905317262762),
+            scale: 1444407
+        )
+        return map
+    }()
+    
+    var body: some View {
+        MapView(map: map, graphicsOverlays: [inputGraphicsOverlay, resultGraphicsOverlay])
+            .onSingleTapGesture { _, tapPoint in
+                self.tapScreenPoint = tapPoint
+            }
+            .overlay(alignment: .center) {
+                // Sets locating indication when currentDrawStatus is true.
+                if currentDrawStatus == true {
+                    ProgressView("Drawing…")
+                        .padding()
+                        .background(.ultraThinMaterial)
+                        .cornerRadius(10)
+                        .shadow(radius: 50)
+                }
+            }
+            .task(id: tapScreenPoint) {
+                guard let tapScreenPoint else { return }
+                currentDrawStatus = true
+                // Sets current draw status to false after process tap point operation
+                // completes.
+                defer { currentDrawStatus = false }
+                await process(at: tapScreenPoint)
+            }
+            .errorAlert(presentingError: $error)
+    }
+    
+    private func process(at tapScreenPoint: Point) async {
+        self.addGraphic(at: tapScreenPoint)
+        await self.calculateViewshed(at: tapScreenPoint)
+    }
+    
+    /// Removes previously tapped location from overlay and draws new dot on tap location.
+    private func addGraphic(at tapPoint: Point) {
```

Comment:
To add the docstring template, use Editor > Structure > Add Documentation, or use its shortcut.

We add docstring for each method, unless it is trivial and self-explanatory (which in most cases, can be refactored away)

Reply:
@yo1995 let me know if this is what you were expecting.